compass/landice/tests/ismip6_forcing/atmosphere/process_smb.py

```diff
@@ -0,0 +1,411 @@
+        """
+        Remap the input ismip6 smb forcing data onto mali mesh
+
+        Parameters
+        ----------
+        input_file: str
+            input smb data on its native polarstereo 8km grid
+        output_file : str
+            smb data remapped on mali mesh
+        mali_mesh_name : str
+            name of the mali mesh used to name mapping files
+        mali_mesh_file : str, optional
+            The MALI mesh file if mapping file does not exist
+        method_remap : str, optional
+            Remapping method used in building a mapping file
+
+        """
+        mapping_file = f"map_ismip6_8km_to_" \
+                       f"{mali_mesh_name}_{method_remap}.nc"
+
+        # check if mapfile exists
+        if not os.path.exists(mapping_file):
+            # build a mapping file if it doesn't already exist
+            self.logger.info(f"Creating a mapping file. "
+                             f"Mapping method used: {method_remap}")
+            build_mapping_file(self.config, self.cores, self.logger,
+                               input_file, mapping_file, mali_mesh_file,
+                               method_remap)
+        else:
+            self.logger.info("Mapping file exists. "
+                             "Remapping the input data...")
+
+        # remap the input data
+        args = ["ncremap",
+                "-i", input_file,
+                "-o", output_file,
+                "-m", mapping_file,
+                "-v", "smb_anomaly"]
+
+        subprocess.check_call(args)
+
+    def rename_ismip6_smb_to_mali_vars(self, remapped_file_temp, output_file):
+        """
+        Rename variables in the remapped source input data
+        to the ones that MALI uses.
+
+        Parameters
+        ----------
+        remapped_file_temp : str
+            temporary ismip6 data remapped on mali mesh
+        output_file : str
+            remapped ismip6 data renamed on mali mesh
+        """
+
+        # open dataset in 20 years chunk
+        ds = xr.open_dataset(remapped_file_temp, chunks=dict(time=20),
+                             engine="netcdf4")
+
+        # build dictionary for ismip6 variables that MALI takes in
+        ismip6_to_mali_dims = dict(
+            time="Time",
+            ncol="nCells")
+        ds = ds.rename(ismip6_to_mali_dims)
+
+        ismip6_to_mali_vars = dict(
+            smb_anomaly="sfcMassBal")
+        ds = ds.rename(ismip6_to_mali_vars)
+
+        # add xtime variable
+        xtime = []
+        for t_index in range(ds.sizes["Time"]):
+            date = ds.Time[t_index]
+            # forcing files do not all match even years, so round up the years
+            # pandas round function does not work for years, so do it manually
+            yr = date.dt.year.values
+            mo = date.dt.month.values
+            dy = date.dt.day.values
+            dec_yr = np.around(yr + (30 * (mo - 1) + dy) / 365.0) # approximate ok
```

Comment:
```suggestion
            # approximate ok
            dec_yr = np.around(yr + (30 * (mo - 1) + dy) / 365.0)
```
I strongly prefer comments before lines rather than on the same line.  If you want to keep it. you need two spaces before the `#`.